File: Codeforces Round 975 (Div. 2)/c.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    int t;
    cin >> t;
    while(t--){
        long long n , k;
        cin >> n >> k;
        long long m = 0 , sum = 0;
        for(long long i=0;i<n;i++){
            long long x;
            cin >> x;
            m = max(m , x);
            sum += x;
        }
        for(long long i=n;i>=1;i--){
            if((sum+k)/i*i < sum) continue;
            if(m > (sum+k)/i*i/i) continue;
            cout << i << endl;
            break;
        }
    }
}

File: Codeforces Round 975 (Div. 2)/c.py
import math

for i in range(int(input())):
    a,coin = [int(x) for x in input().split()]
    n = [int(x) for x in input().split()]
    p = max(n)
    total = sum(n)
    q = a*p - total
    if coin >= q:
        print(min(a , coin + total))

    else:
        if total % p == 0 or p - total % p <= coin:
            print((total+coin)//p)
        else:
            print(min((total+coin - (total+coin)%p)//p , 
                      (total+coin - (total+coin)%a)//a))



        

